### ISSUE TITLE
Calculating the exchange rate

### ISSUE THREAD

**Issue by shinsyotta** [olehmisar/aztec-amm#7]
It looks like the exchange rate is calculated on the front end. Is this going to lead to lots of race condition-related failures?

**Comment by olehmisar** [comment]
Yes, it is going to lead to lots of race condition-related failures. This is a tech demo. Implementing race-condition free logic is complicated because of private <-> public context communication. A PR is welcome.
